noisemodelling-pathfinder/src/main/java/org/noise_planet/noisemodelling/pathfinder/ComputeRaysOut.java

```diff
@@ -0,0 +1,215 @@
+ * <p>
+ * This version is developed at French IRSTV Institute and at IFSTTAR
+ * (http://www.ifsttar.fr/) as part of the Eval-PDU project, funded by the
+ * French Agence Nationale de la Recherche (ANR) under contract ANR-08-VILL-0005-01.
+ * <p>
+ * Noisemap is distributed under GPL 3 license. Its reference contact is Judicaël
+ * Picaut <judicael.picaut@ifsttar.fr>. It is maintained by Nicolas Fortin
+ * as part of the "Atelier SIG" team of the IRSTV Institute <http://www.irstv.fr/>.
+ * <p>
+ * Copyright (C) 2011 IFSTTAR
+ * Copyright (C) 2011-2012 IRSTV (FR CNRS 2488)
+ * <p>
+ * Noisemap is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option) any later
+ * version.
+ * <p>
+ * Noisemap is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License along with
+ * Noisemap. If not, see <http://www.gnu.org/licenses/>.
+ * <p>
+ * For more information, please consult: <http://www.orbisgis.org/>
+ * or contact directly:
+ * info_at_ orbisgis.org
+ */
+package org.noise_planet.noisemodelling.pathfinder;
+
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Way to store data computed by threads.
+ * Multiple threads use one instance.
+ * This class must be thread safe
+ *
+ * @author Nicolas Fortin
+ * @author Pierre Aumond
+ */
+public class ComputeRaysOut implements IComputeRaysOut {
+    public List<PropagationPath> propagationPaths = Collections.synchronizedList(new ArrayList<PropagationPath>());
+
+    //public PropagationProcessPathData genericMeteoData;
+    public PropagationProcessData inputData;
+
+    public ComputeRaysOut(boolean keepRays, PropagationProcessData inputData) {
+        this.keepRays = keepRays;
+        this.inputData = inputData;
+    }
+
+    public ComputeRaysOut(boolean keepRays) {
+        this.keepRays = keepRays;
+    }
+
+    public boolean keepRays = true;
+    public AtomicLong rayCount = new AtomicLong();
+    public AtomicLong nb_couple_receiver_src = new AtomicLong();
+    public AtomicLong nb_obstr_test = new AtomicLong();
+    public AtomicLong nb_image_receiver = new AtomicLong();
+    public AtomicLong nb_reflexion_path = new AtomicLong();
+    public AtomicLong nb_diffraction_path = new AtomicLong();
+    public AtomicInteger cellComputed = new AtomicInteger();
+    // private static final double angle_section = (2 * Math.PI) / PropagationProcessPathData.DEFAULT_WIND_ROSE.length;
+
+
+   /* public static int getRoseIndex(Coordinate receiver, Coordinate source) {
```

Comment:
maybe remove this if it is not used anymore ?

Reply:
Yes ! I do it directly in my fork ?